components/usage/pkg/apiv1/billing.go

```diff
@@ -165,6 +177,61 @@ func (s *BillingService) CreateStripeCustomer(ctx context.Context, req *v1.Creat
 	}, nil
 }
 
+func (s *BillingService) CreateStripeSubscription(ctx context.Context, req *v1.CreateStripeSubscriptionRequest) (*v1.CreateStripeSubscriptionResponse, error) {
+	attributionID, err := db.ParseAttributionID(req.GetAttributionId())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid attribution ID %s", attributionID)
+	}
+
+	if req.CustomerId == "" {
+		return nil, status.Error(codes.InvalidArgument, "Invalid customer ID")
+	}
+
+	_, err = s.stripeClient.SetDefaultPaymentForCustomer(ctx, req.CustomerId, req.SetupIntentId)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to set default payment for customer ID %s", req.CustomerId)
+	}
+
+	stripeCustomer, err := s.stripeClient.GetCustomer(ctx, req.CustomerId)
+	if err != nil {
+		return nil, err
+	}
+
+	priceIdentifier := getPriceIdentifier(attributionID, stripeCustomer, s)
+
+	subscription, err := s.stripeClient.CreateSubscription(ctx, req.CustomerId, priceIdentifier)
+	if err != nil {
+		return nil, status.Errorf(codes.Aborted, "Failed to create subscription with customer ID %s", req.CustomerId)
+	}
+
+	return &v1.CreateStripeSubscriptionResponse{
+		Subscription: &v1.StripeSubscription{
+			Id: subscription.ID,
+		},
+	}, nil
+}
+
+func getPriceIdentifier(attributionID db.AttributionID, stripeCustomer *stripe_api.Customer, s *BillingService) string {
+	priceIdentifier := ""
+	if attributionID.IsEntity("team") {
+		if stripeCustomer.Currency == "EUR" {
+			priceIdentifier = s.stripePrices.TeamUsagePriceIDs.EUR
+		}
+		if stripeCustomer.Currency == "USD" {
+			priceIdentifier = s.stripePrices.TeamUsagePriceIDs.USD
+		}
```

Comment:
What if the `stripeCustomer.Currency` is neither EUR, nor USD?

Use a switch here to handle that case.

Reply:
If it's neither, then it's an empty string.